### ISSUE TITLE
Error in Using Nvidia DIGITS trained Models in OpenCV

### ISSUE THREAD

**Issue by Yasin40** [NVIDIA/DIGITS#2036]
I Trained my model using DIGITS ( NVCaffe) and I use it in opencv similar to opencv example that use .prototxt & .caffemodel and i tested it and work good. But when I use Models that trained by digits i got this error:

OpenCV Error: Assertion failed (pbBlob.raw_data_type() == 
caffe::FLOAT16) in blobFromProto, file 
/opt/opencv/modules/dnn/src/caffe/caffe_importer.cpp, line 242
Exception:  /opt/opencv/modules/dnn/src/caffe/caffe_importer.cpp:242: 
error: (-215) pbBlob.raw_data_type() == caffe::FLOAT16 in function 
blobFromProto
help me to solve it! this problem mentioned many times ago without answer! http://answers.opencv.org/question/177086/unable-to-use-caffe-model-trained-in-nvidia-digits-in-opencv-dnn-code/

**Comment by nnowaker** [comment]
same question here ! 
anyone can help us ?!

**Comment by nnowaker** [comment]
finally, I found the problem! it's a version problem I have digits 6.1.1 working with nvcaffe 0.17.0 for training which is not compatible with previous Caffe and OpenCv libraries ! you have to downgrade NvCaffe to version 0.15.14 and it will open with OpenCv easily!

**Comment by sagartesla** [comment]
OpenCV DNN model expect caffemodel in BVLC format. But, NVCaffe stores the caffe model in more efficient format which different than BVLC Caffe. If you want model compatible with both BVLC/Caffe as well as NVcaffe.

Add this flag in solver.prototxt store_blobs_in_old_format = true

Please read the DIGITS NVCaffe Documentation:
https://docs.nvidia.com/deeplearning/dgx/caffe-user-guide/index.html#storeblobsold

**Comment by mprat** [comment]
This option is not exposed in DIGITS 6.1.1 and is making model debugging nearly impossible. Can you please recommend a solution?

**Comment by sagartesla** [comment]
@MARS-Rovers @Samvid95 

Hi, 

As i have explained in my last reply.
Before starting your training, You have to modify your solver.prototxt

add following line:
store_blobs_in_old_format = true

And then do finish network training. 
I am sure that you will not face the error which your facing in opencv.

example: 
Consider this solver.prototxt

# The train/test net protocol buffer definition
net: "examples/mnist/lenet_train_test.prototxt"
# test_iter specifies how many forward passes the test should carry out.
# In the case of MNIST, we have test batch size 100 and 100 test iterations,
# covering the full 10,000 testing images.
test_iter: 100
# Carry out testing every 500 training iterations.
test_interval: 500
# The base learning rate, momentum and the weight decay of the network.
base_lr: 0.01
momentum: 0.9
weight_decay: 0.0005
# The learning rate policy
lr_policy: "inv"
gamma: 0.0001
power: 0.75
# Display every 100 iterations
display: 100
# The maximum number of iterations
max_iter: 10000
# snapshot intermediate results
snapshot: 5000
snapshot_prefix: "examples/mnist/lenet"
# solver mode: CPU or GPU
solver_mode: GPU
# stores blobs in an old, less efficient BVLC-compatible format
**store_blobs_in_old_format = true**


Let me also know the nvcaffe version you are using in your current DIGITS setup.

**Comment by MARS-Rovers** [comment]
@sagartesla 
Where is the solver.prototxt stored? I've checked the "jobs" directory, and I only see a "model.caffemodel", "original.prototxt", and a "status.pickle". I understand that we have to add the "store_blobs_in_old_format = true" line to the solver.prototxt, but I can't find where that file is stored within the docker container (I'm using "sudo docker exec -it digits bash" to view the files within the digits container).

**Comment by sagartesla** [comment]
@MARS-Rovers 


Ok. Let me understand one thing. Are you running some example which is available in NVIDIA-DIGITS docker. ?

As you said, solver.prototxt does not exist before training start. 
1. Check which script are you executing ?
- sudo docker exec -it digits bash this will open bash terminal
But, to start training process, you must be executing some .sh file or python file

2. Check that .sh or python file
- This .sh file must be having all the solver related parameters inside hardcoded.
And these are passed to python file where solver.prototxt, train.prototxt etc files are generated using pycaffe interface

3. Just explore that job file
- the file which is executing job


I hope this answer will help you.
Otherwise just share me directory structure and what all commands you are executing to start training.

**Comment by rkbmids** [comment]
@sagartesla i'm creating a model through the nvidia digits UI. How do I find the model executable and modify this parameter? I've modified solver.prototxt in the docker container but the changes don't show up in the solver.prototxt file that shows in the nvidia digits ui.
